File: FC.Shared/Entities/UserLocalization.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FC.Shared.Entities
{
    public class UserLocalization
    {
        public UserLocalization(CultureInfo culture, RegionInfo region)
        {
            CultureIsoName = culture.TwoLetterISOLanguageName + "-"+region.TwoLetterISORegionName;
            CultureMoneySign = region.CurrencySymbol;
            RegionName = region.DisplayName;
            RegionEnglishName = region.EnglishName;
            CurrencyNativeName = region.CurrencyNativeName;
            CultureCountryName = region.DisplayName;
            CultureDateSeparator = culture.DateTimeFormat.DateSeparator;
            TimeSeparator = culture.DateTimeFormat.TimeSeparator;
            CultureCurrencySeparator = culture.NumberFormat.CurrencyDecimalSeparator;
            NumberDecimalSeparator = culture.NumberFormat.NumberDecimalSeparator;
            CurrencyCultureDecimalDigits = culture.NumberFormat.CurrencyDecimalDigits;
            NumberDecimalDigits = culture.NumberFormat.NumberDecimalDigits;
            PositiveSign = culture.NumberFormat.PositiveSign;
            NegativeSign = culture.NumberFormat.NegativeSign;
            ISOCurrencySymbol = region.ISOCurrencySymbol;
            TwoLetterCountryName = region.TwoLetterISORegionName;
            ThreeLetterCountryName = region.ThreeLetterISORegionName;
                
        }
        public string TwoLetterCountryName { get; set; }
        public string ThreeLetterCountryName { get; set; }

        public string ISOCurrencySymbol { get; set; }
        public string NegativeSign { get; set; }
        public string PositiveSign { get; set; }
        public string TimeSeparator { get; set; }
        public string CurrencyNativeName { get; set; }
        public string RegionEnglishName { get; set; }
        public string RegionName { get; set; }
        public string CultureIsoName { get; set; }
        public string CultureCountryName { get; set;}
        public string CultureMoneySign { get; set; }
        public string CultureDateSeparator { get; set; }
        public string CultureCurrencySeparator { get; set; }
        public int CurrencyCultureDecimalDigits { get; set; }
        public int NumberDecimalDigits { get; set; }
        public string NumberDecimalSeparator { get; set; }

    }
}

File: FC.WebMVC/Scripts/Shared/Models/UserLocalization.ts
module FC.Shared.Models {
    import CM = FC.Core.CoreModel;
    import INT = FC.Shared.Interfaces;
    import MODELS = FC.Shared.Models;
    import MODULES = FC.Modules;
    export class UserLocalization implements INT.IUserLocalization {
        constructor(UL: INT.IUserLocalization) {
            this.ISOCurrencySymbol = UL.ISOCurrencySymbol;
            this.NegativeSign = UL.NegativeSign;
            this.PositiveSign = UL.PositiveSign;
            this.TimeSeparator = UL.TimeSeparator;
            this.CurrencyNativeName = UL.CurrencyNativeName;
            this.RegionEnglishName = UL.RegionEnglishName;
            this.RegionName = UL.RegionName;
            this.CultureIsoName = UL.CultureIsoName;
            this.CultureCountryName = UL.CultureCountryName;
            this.CultureMoneySign = UL.CultureMoneySign;
            this.CultureDateSeparator = UL.CultureDateSeparator;
            this.CultureCurrencySeparator = UL.CultureCurrencySeparator;
            this.CurrencyCultureDecimalDigits = UL.CurrencyCultureDecimalDigits;
            this.NumberDecimalDigits = UL.NumberDecimalDigits;
            this.NumberDecimalSeparator = UL.NumberDecimalSeparator;
            this.TwoLetterCountryName = UL.TwoLetterCountryName;
            this.ThreeLetterCountryName = UL.ThreeLetterCountryName;
        }
        ISOCurrencySymbol: string;
        NegativeSign: string;
        PositiveSign: string;
        TimeSeparator: string;
        CurrencyNativeName: string;
        RegionEnglishName: string;
        RegionName: string;
        CultureIsoName: string;
        CultureCountryName: string;
        CultureMoneySign: string;
        CultureDateSeparator: string;
        CultureCurrencySeparator: string;
        CurrencyCultureDecimalDigits: Number;
        NumberDecimalDigits: Number;
        NumberDecimalSeparator: string;
        TwoLetterCountryName: string;
        ThreeLetterCountryName: string;
    }
}
